OpenAlex JSON.js

```diff
@@ -184,11 +182,22 @@ function parseIndividual(data) {
 		}
 		item.attachments.push({ url: data.best_oa_location.pdf_url, title: version, mimeType: "application/pdf" });
 	}
+	if (data.best_oa_location && data.best_oa_location.landing_page_url) {
+		// Don't save DOI urls
+		if (!data.best_oa_location.landing_page_url.includes("doi.org")) item.url = data.best_oa_location.landing_page_url;
+	}
+	else if (data.primary_location && data.primary_location.landing_page_url) {
+		// Don't save DOI urls
+		if (!data.primary_location.landing_page_url.includes("doi.org")) item.url = data.primary_location.landing_page_url;
+	}
 	let tags = data.keywords;
 	for (let tag of tags) {
 		item.tags.push(tag.display_name || tag.keyword);
 	}
-	item.extra = "OpenAlex: " + data.ids.openalex;
+	let openAlexID = data.ids.openalex.match(/W\d+$/i)[0];
+	item.setExtra("OpenAlex", openAlexID);
+	item.libraryCatalog = "OpenAlex";
```

Comment:
I don't know what you settled on in the past, but my suggestion is to focus on the identifier. Sure, OpenAlex stores them with the URL, but even their documentation shows that the actual identifier for works is just the part that starts with W. In the end, it's the same reason why we canonically store just the identifier for arXiv or DOI, even though you can build the URLs with them. Lastly, it's just more terse and legible IMO. In any event, the translator supports searching with both.

Reply:
The reason we used the version with domain is that that's defined as the OpenAlex ID by OpenAlex:
https://docs.openalex.org/how-to-use-the-api/get-single-entities#the-openalex-id

> The OpenAlex ID is the primary key for all entities. It's a URL shaped like this: https://openalex.org/<OpenAlex_key>. Here's a real-world example:
> 
> https://openalex.org/W2741809807

I'm not supper opposed to handling this differently, and I can see this _looks_ like it's inconsistent, but for DOIs, PMIDs, and arXiv IDs, the ID explicitly does not include the URL schema, so this isn't the same.

Reply:
That's what we would be doing, but I'm not convinced going with the key is the right choice. Legibility is obviously a matter of taste, but storing identifiers in a format in which they're self-identifying seems like a good idea: DOIs and ISBNs are to by virtue of their format; arXiv IDs canonically include the arxiv: namespace (yes, we do store the openAlex: label in Extra, but that's as part of a key:value pair, not as part of the identifier). PMIDs are a constant problem. OpenAlex keys are marginally better because of the W, but only marginally so.